cmd/lotus/daemon.go

```diff
@@ -639,10 +638,12 @@ func slashConsensus(a lapi.FullNode, p string, from string) error {
 	}
 	for block := range blocks {
 		log.Infof("deal with block: %d, %v, %s", block.Height, block.Miner, block.Cid())
-		if otherBlock, extraBlock, err := slashFilterMinedBlock(ctx, sf, a, block); err != nil {
-			if otherBlock == nil {
-				continue
-			}
+		otherBlock, extraBlock, fault, err := slashFilterMinedBlock(ctx, sf, a, block)
+		if err != nil {
+			log.Errorf("slash detector errored: %s", err)
+			continue
+		}
+		if fault {
 			log.Errorf("<!!> SLASH FILTER ERROR: %s", err)
```

Comment:
I think we are losing information here.  Before this change the error contained information about the slashing condition being hit and the problem blocks.  Now I think this err will just be nil

Reply:
Good catch, thank you -- fixing.